helpers/database-import-scripts/uniprot/convert_gbk.py

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env python3
+
+import argparse
+import logging
+import os
+import re
+import sys
+import textwrap
+
+logging.basicConfig(level=logging.INFO)
+
+
+def main(gbk_file, outfile, taxonomy_file):
+    accession = os.path.basename(gbk_file).replace(".gbk", "")
+    species, lineage, taxid = ingest_taxonomy_file(taxonomy_file, accession)
+    if not species:
+        logging.info("Uniprot output file is not generated because species is unknown: {}".format(accession))
+        sys.exit()
+    if lineage.endswith("s__"):
+        logging.info("NCBI species for genome {} is unknown. Uniprot import file will not be generated.".
+                     format(accession))
+    process_file(gbk_file, outfile, taxid, species, lineage)
+
+    
+def ingest_taxonomy_file(taxonomy_file, accession):
+    species = lineage = taxid = ""
+    with open(taxonomy_file, "r") as file_in:
+        for line in file_in:
+            if line.startswith(accession):
+                _, species, lineage, _, taxid, _, _ = line.strip().split("\t")
+                return species, lineage, taxid
+    return species, lineage, taxid
+
+
+def format_lineage(taxonomy_lineage):
+    # Split the lineage into segments
+    segments = taxonomy_lineage.split(';')
+    
+    # Remove empty segments from the end
+    while segments and re.match(r'[a-z]__$', segments[-1]):
+        segments.pop()
+
+    # Remove the last known (named) segment
+    segments.pop()
+
+    # Extract the names from the segments and format the output
+    formatted_lineage = '; '.join([segment.split('__')[1] for segment in segments])
+
+    # Print the formatted lineage with proper indentation and line breaks
+    indent = " " * 12  # Adjust the number of spaces for your desired indentation
+    lines = []
+    line_length = 0
+    for segment in formatted_lineage.split("; "):
+        if not lines or line_length + len(segment) + 2 > 68:
+            lines.append(segment + "; ")
+            line_length = len(segment) + 2
+        else:
+            lines[-1] += segment + "; "
+            line_length += len(segment) + 2
+
+    formatted_output = "\n".join([indent + line.rstrip(" ") for line in lines])
+    formatted_output = re.sub(r';$', '', formatted_output) + "."
+    if "; ;" in formatted_output:
+        logging.info("Taxonomy lineage is missing fields: {}".format(formatted_output))
+    return formatted_output
+
+
+def process_file(gbk_file, outfile, taxid, species, lineage):
```

Comment:
Why didn't you use biopython to read the gbk?

Reply:
Probably should have